packages/core/src/layout/SignInPage/SignInPage.tsx

```diff
@@ -69,3 +77,86 @@ export const SignInPage = ({
     </Page>
   );
 };
+
+export const SingleSignInPage = ({
+  onResult,
+  provider,
+  auto,
+}: SingleSignInPageProps) => {
+  const classes = useStyles();
+  const authApi = useApi(provider.apiRef);
+  const configApi = useApi(configApiRef);
+
+  const [retry, setRetry] = useState<{} | boolean | undefined>(auto);
+  const [error, setError] = useState<Error>();
+
+  useEffect(() => {
+    const login = async () => {
+      const identity = await authApi.getBackstageIdentity({
+        instantPopup: true,
+      });
+
+      const profile = await authApi.getProfile();
+      onResult({
+        userId: identity!.id,
+        profile: profile!,
+        getIdToken: () => {
+          return authApi.getBackstageIdentity().then(i => i!.idToken);
+        },
+        signOut: async () => {
+          await authApi.signOut();
+        },
+      });
+    };
+
+    if (retry) {
+      login().catch(setError);
+    }
+  }, [onResult, authApi, retry]);
+
+  return (
+    <Page themeId="home">
+      <Header title={configApi.getString('app.title')} />
+      <Content>
+        <Grid
+          container
+          justify="center"
+          spacing={2}
+          component="ul"
+          classes={classes}
+        >
+          <GridItem>
+            <InfoCard
+              variant="fullHeight"
+              title={provider.title}
+              actions={
+                <Button
+                  color="primary"
+                  variant="outlined"
+                  onClick={() => setRetry({})}
```

Comment:
Just to get a new instance? Maybe something like `setRetry(x => x + 1)` for a number would have been neater

Reply:
**I mean** *we could do* **that**<i>, but it</i><b>'s like</b>*ly that* **your** *version could result in a bit more code that in my* **opinion** *is not necessarily more* **man***agable* :trollface:

More srsly I don't mind the extra objects, it's not a tight loop if that's the concern. Using numbers requires a bit more code since `undefined | boolean` needs to be represented as well, as we're passing that in as the initial value from the `auto` prop. Simple enough with a `Number(auto)`, but a bit more code and not sure if it's more clear. Happy to switch it around if people find it more readable tho 😅

Reply:
Haha. Nah no performance concern at all. Only a matter of me not having seen that pattern ever, and it kinda relies on some level of deeper understanding of things :) So for the principle of least surprise for the reader, it felt clearer with something that clearly changes.

Reply:
If one assumes that the code works, then this could be considered a learning opportunity! :grin: